diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2a62fde..ee8f90c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -152,6 +152,11 @@ endif()
 # Other sources
 add_subdirectory(src)
 
+add_executable(ecky_ftetwild_worker ecky/ftetwild_worker.cpp)
+target_compile_features(ecky_ftetwild_worker PUBLIC cxx_std_17)
+target_link_libraries(ecky_ftetwild_worker PUBLIC ${PROJECT_NAME} warnings::all)
+set_target_properties(ecky_ftetwild_worker PROPERTIES CXX_EXTENSIONS OFF)
+
 # Compile extras only if this is a top-level project
 if(FLOAT_TETWILD_TOPLEVEL_PROJECT)
     # Unit tests
diff --git a/src/Mesh.hpp b/src/Mesh.hpp
index 662ee08..fd5a8bc 100644
--- a/src/Mesh.hpp
+++ b/src/Mesh.hpp
@@ -12,6 +12,7 @@
 
 
 // #ifdef FLOAT_TETWILD_USE_TBB
+#include <cstdint>
 // #include <tbb/concurrent_vector.h>
 // #endif
 
@@ -152,7 +153,7 @@ namespace floatTetWild {
         std::array<char, 4> is_surface_fs = {{NOT_SURFACE, NOT_SURFACE, NOT_SURFACE, NOT_SURFACE}};
         std::array<char, 4> is_bbox_fs = {{NOT_BBOX, NOT_BBOX, NOT_BBOX, NOT_BBOX}};
         std::array<int, 4> opp_t_ids = {{OPP_T_ID_UNKNOWN, OPP_T_ID_UNKNOWN, OPP_T_ID_UNKNOWN, OPP_T_ID_UNKNOWN}};
-        std::array<char, 4> surface_tags = {{0, 0, 0, 0}};
+        std::array<std::int32_t, 4> surface_tags = {{0, 0, 0, 0}};
 
         Scalar quality = 0;
         Scalar scalar = 0;
